src/shader_uniforms.rs

```diff
@@ -70,96 +69,118 @@ impl UniformType for SUMat4x4 {}
 
 impl GlGraphics {
     /// Try to get uniform from the current shader of a given name.
-    pub fn get_uniform<T : UniformType + ?Sized>(&self, name : &str) -> Option<ShaderUniform<T>> {
-        self.get_current_program().and_then( |p| {
-            unsafe {
-                let c_source = CString::new(name).ok();
-                c_source.and_then(|name| {
-                    let uniform = match gl::GetUniformLocation(p, name.as_ptr()) {
-                        -1 => None,
-                        location => {
-                            Some(ShaderUniform{
-                                location : location,
-                                phantom : PhantomData,
-                            })
-                        },
-                    };
-                    drop(name);
-                    uniform
-                })
-            }
+    pub fn get_uniform<T: UniformType + ?Sized>(&self, name: &str) -> Option<ShaderUniform<T>> {
+        self.get_current_program().and_then(|p| unsafe {
+            let c_source = CString::new(name).ok();
+            c_source.and_then(|name| {
+                let uniform = match gl::GetUniformLocation(p, name.as_ptr()) {
+                    -1 => None,
+                    location => Some(ShaderUniform {
+                        location,
+                        phantom: PhantomData,
+                    }),
+                };
+                drop(name);
+                uniform
+            })
         })
     }
 }
 
 impl ShaderUniform<SUFloat> {
     /// Set the value of the float uniform.
-    pub fn set(&self, gl : &GlGraphics, value : f32) {
-        gl.get_current_program().map(|p| {
-            unsafe {gl::ProgramUniform1f(p, self.location, value)}
-        });
+    pub fn set(&self, gl: &GlGraphics, value: f32) {
```

Comment:
these and following:
i am unsure if silently failing is the desired behaviour, i assume that there is probably already an earlier check that catches the lack of a current_program.